scrapy/commands/genspider.py

```diff
@@ -5,6 +5,7 @@
 
 from importlib import import_module
 from os.path import join, dirname, abspath, exists, splitext
+from urllib.parse import urlparse
```

Comment:
```suggestion
from six.moves.urllib.parse import urlparse
```
Tests are failing in py2 because `urllib.parse` is py3-only.